### PR TITLE
Connections wording + Gdrive warning improvement proposal

### PR DESCRIPTION
**Pull request by spolu**
Before:
Screenshot from 2023-12-15 09-33-38

After:
Screenshot from 2023-12-18 11-46-38

Screenshot from 2023-12-18 11-47-06
Screenshot from 2023-12-18 11-47-00
Screenshot from 2023-12-18 11-46-55
Screenshot from 2023-12-18 11-46-50
Screenshot from 2023-12-18 11-46-45

### CODE REVIEW

**Comment by Duncid** [comment]
# My understanding

You removed "Limitation" added a new warning step before connection to any 3rd party (re-using "limitations" and more).

**This is not possible in drop down any more, needs to go to a side modal!**
image

File: front/lib/connector_providers.ts [hunk 1]
```diff
@@ -27,7 +28,8 @@ export const CONNECTOR_CONFIGURATIONS: Record<
     isBuilt: true,
     logoPath: "/static/notion_32x32.png",
     description:
-      "Authorize granular access to your company's Notion workspace, by top-level pages.",
+      "Authorize granular access to your company's Notion workspace, by top-level pages. External files and content behind links are not indexed.",
```
Comment (Duncid) [inline front/lib/connector_providers.ts:31]: "Content indexed will be made available to your workspace regardless of authorizations in Notion" ?

File: front/lib/connector_providers.ts [hunk 2]
```diff
@@ -37,7 +39,8 @@ export const CONNECTOR_CONFIGURATIONS: Record<
     isBuilt: true,
     logoPath: "/static/google_drive_32x32.png",
     description:
-      "Authorize granular access to your company's Google Drive, by drives and folders. Supported files include GDocs, GSlides, and .txt files.",
+      "Authorize granular access to your company's Google Drive, by drives and folders. Supported files include GDocs, GSlides, and .txt files. Email us for .pdf indexation.",
+    limitations: "Files with more than 750KB of extracted text are ignored.",
```
Comment (Duncid) [inline front/lib/connector_providers.ts:43]: Now the we have more room, I would shortly explain why and be clearer on "extracted" VS "file size".

"Dust will read all files including large PDFs but will ignore them when extracted text content is bigger than 750Kb (~XXX pages of texts)"

**Comment by spolu** [comment]
I moved most of limitations in the description of the connection assuming no-one would really read them + added the Important section on the dropdown that existed already which has a more decent chance of being read.

File: front/lib/connector_providers.ts [hunk 3]
```diff
@@ -47,7 +50,8 @@ export const CONNECTOR_CONFIGURATIONS: Record<
     isBuilt: true,
     logoPath: "/static/slack_32x32.png",
     description:
-      "Authorize granular access to your Slack workspace on a channel-by-channel basis.",
+      "Authorize granular access to your Slack workspace on a channel-by-channel basis. External files and content behind links are not indexed.",
```
Comment (Duncid) [inline front/lib/connector_providers.ts:53]: Maybe same here — will ignore slack's authorisation

**Review by Duncid — APPROVED** [review]

**Comment by Duncid** [comment]
LGTM — I would maybe be less formal and drop the "acknowledge" and move to upper button size, but very nit.

**Review by PopDaph — APPROVED** [review]

**Comment by spolu** [comment]
> LGTM — I would maybe be less formal and drop the "acknowledge" and move to upper button size, but very nit.

I think this wording is required for google drive. I'll remove it for other connectors and make the button bigger :+1:

File: front/pages/w/[wId]/builder/data-sources/managed.tsx [hunk 4]
```diff
@@ -432,11 +564,8 @@ export default function DataSourcesView({
                           }
 
                           if (isDataSourceAllowedInPlan && ds.isBuilt) {
-                            await handleEnableManagedDataSource(
-                              ds.connectorProvider as ConnectorProvider,
-                              ds.setupWithSuffix
-                            );
-                          } else if (isDataSourceAllowedInPlan && !ds.isBuilt) {
+                            setShowConfirmConnection(ds);
+                          } else if (!ds.isBuilt) {
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/pages/w/[wId]/builder/data-sources/managed.tsx]: reverted as per IRL :+1:

**Comment by spolu** [comment]
@Duncid I presumed by "upper button size" you meant `md`

**Pull request merged**